File: assp/elem/07_mult_table.py
""" Elementary 7:
    Write a program that prints a multiplication
    table for numbers up to 12
"""

from tabulate import tabulate

mult_dict = {}
nums = range(1,13)

for i in nums:
    temp_list = []
    for j in nums:
        temp_list.append(i*j)
    mult_dict[i] = temp_list

print("Multiplication Table (12x12)")
print(tabulate(mult_dict, headers="keys", showindex=nums))

File: assp/elem/07_mult_table.c
// Elementary 7: Multiplication Table for numbers up to 12

#include <stdio.h>

int main(void)
{
    int tab;
    int num = 12;

    printf("Multiplication Table for 12:\n");
    for (int i = 1; i <=12; i++)
    {
	tab = num * i;
	printf("%d * %2d = %2d\n", num, i, tab);
    }

    return 0;
}
